### PR TITLE
chore: Add docs for user interface and export the interface

### PR DESCRIPTION
**Pull request by florian-richter**
Please provide a description of what your change does and why it is needed.

Closes SAP/cloud-sdk-backlog#40

@jjtang1985 @marikaner Should the generated docs be checked in as well? There are quite a lot of changes as some dependencies have changed and references to dependency types now have changes throughout the docs.

### CODE REVIEW

**Comment by jjtang1985** [comment]
@florian-richter , not sure whether I got the question correctly.
The new doc will be updated when we have a new release like [this](https://github.com/SAP/cloud-sdk-js/actions/runs/1059843907).

File: packages/core/src/connectivity/scp-cf/index.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export {
   isIdenticalTenant
 } from './tenant';
 export {
+  User,
```
Comment (jjtang1985) [inline packages/core/src/connectivity/scp-cf/index.ts:26]: It seems, the `User` is not used 🤔

**Review by jjtang1985 — APPROVED** [review]
LGTM

**Comment by marikaner** [comment]
> Please provide a description of what your change does and why it is needed.
> 
> Closes [SAP/cloud-sdk-backlog#40](https://github.com/SAP/cloud-sdk-backlog/issues/40)
> 
> @jjtang1985 @marikaner Should the generated docs be checked in as well? There are quite a lot of changes as some dependencies have changed and references to dependency types now have changes throughout the docs.

Nope.

**Pull request merged**